shapely/geos.py

```diff
@@ -140,7 +144,7 @@ def load_dll(libname, fallbacks=None, mode=DEFAULT_MODE):
     free.restype = None
 
 elif sys.platform == 'win32':
-    if os.getenv('CONDA_PREFIX', ''):
+    if exists_conda_env():
```

Comment:
Can you add an additional `and ..` check here that the file actually exists? 
That should solve the problem of "wheel in a conda env" (at least for the case where the conda geos package is not installed ..)

Reply:
I don't quite understand @jorisvandenbossche . Which file, the wheel's GEOS DLL?

I still feel like we're setting users up for failure by enabling them to `conda install gdal; pip install shapely` and then expect these to work together in the same python program. But not using the DLL in the wheel would be the worst thing that could happen, for sure.

Reply:
This branch sets the path to `os.path.join(sys.prefix, 'Library', 'bin', 'geos_c.dll')`, without checking that this actually exists (which is indeed not necessarily the case if installed with pip). So I would personally at least check that the file actually exists.

Reply:
> I still feel like we're setting users up for failure by enabling them to `conda install gdal; pip install shapely` and then expect these to work together in the same python program. 

Agree 100%. They should use conda into install all three – GEOS, GDAL, and Shapely. The spread of `is CONDA_PREFIX` allover the codebase is a bit ugly and symptomatic of code smell on the Conda side, not a failure of any of the libraries.

Reply:
If you really, really don't want to support pip-installation into conda envs (even though my personal experience is that it just works with the patch above), please at least consider checking if the dll/so file exists and emitting a clearer error if it doesn't ("you appear to be in a conda environment but some files are missing from shapely, please note that shapely does not support pip installs in conda environments" or something similar).

Reply:
We already enable this for linux and mac, so rephrasing my earlier question: is there a reason that on Windows this would be less robust? 

People are going to do this anyway. First, because there are legitimate reasons to do so (eg using conda only as environment manager, installing nightly wheels if they would be available (packages like numpy, scikit-learn, pandas provide those, and those are regularly installed in conda envs), ..). And second because people are simply not aware, and the internet is full of instructions to do `pip install shapely`. 
So personally, I think we should either try to ensure it works properly, or either try to raise an informative error message. But for both options, we will need to detect this case.

Reply:
Right, let's continue to isolate this case.

If shapely didn't dlopen libgeos (via ctypes) we wouldn't have this problem. This is what makes the flaky shapely/gdal/fiona + libgeos combination different from, for example, the stable numpy/scipy + libgfortran combination. I don't blame the conda design at all.

We're planning a rewrite of shapely to eliminate ctypes and which will delete all the code we're discussing now. Let's try to make the best of a bad situation and solve this particular issue. I'm going to stop opposing and complaining about pip installing shapely wheels into a conda env. In return, I'm going to rely on the community for helping in supporting users who do this.